wger/nutrition/models.py

```diff
@@ -137,10 +138,17 @@ def get_nutritional_values(self):
             for key in result['percent'].keys():
                 result['percent'][key] = \
                     result['total'][key] * ENERGY_FACTOR[key][unit] / energy * 100
-
+        
         # Per body weight
         if self.user.userprofile.weight:
-            weight = Decimal(self.user.userprofile.weight)
+            # Determine closest entry by date
+            target = self.creation_date
+            closest_entry = None
+            for entry in WeightEntry.objects.filter(user = self.user).order_by('date').iterator():
+                if not closest_entry or \
+                (abs(entry.date - self.creation_date) <= abs(closest_entry.date - self.creation_date)):
+                    closest_entry = entry
+            weight = Decimal(closest_entry.weight)
```

Comment:
Better than iterating through all the entries, I'd do something [similar to this](https://stackoverflow.com/questions/15855715/filter-on-datetime-closest-to-the-given-datetime). I also think it's possible to replace all the try-excepts by using ``[....].order_by('date').first()``, but I'm not 100% sure